app/policies/article_policy.rb

```diff
@@ -77,6 +77,27 @@ def feed?
     true
   end
 
+  # Does the user already have existing articles?  Can they create an article?
+  #
+  # @return [TrueClass] They have existing published articles OR can create new ones.
+  # @return [FalseClass] They do not have published articles NOR can they create new ones.
+  #
+  # @note As part of our aspirations to only show users what is relevant to them and "hiding" what
+  #       is not, this method will help us with the edge case of "should we show the user a
+  #       dashboard listing of posts?"
+  #
+  # @note This policy method is a bit different.  It is strictly meant to return true or false.
+  #       Other policies might raise exceptions, but the purpose of this method is for conditional
+  #       rendering.
+  def has_existing_articles_or_can_create_new_ones?
+    require_user!
+    return true if user.articles.published.exists?
+
+    create?
+  rescue ApplicationPolicy::NotAuthorizedError
+    false
+  end
```

Comment:
I think I understand your line of thinking here. But, adding the line `return true if user.articles.published.exists?` in the `create?` method seems like the way to go to me. Unless there is a world in which a user cannot create posts even though they have articles published.

for the conditional rendering piece, maybe we can create a headless policy explicitly for the dashboard for example

Reply:
I'll add a comment to clarify.  As I'm very specifically mean these two separate clauses

Reply:
> This handles the case in which a user has lost the ability to create posts (e.g. we've toggled on the feature limiting posts to admins only) but they have at least one published post.  In that case we want to show them things like "their posts's analytics" or a dashboard of their published posts.

From the comment.